dotnet/src/Extensions/Planning.ActionPlanner/ActionPlanner.cs

```diff
@@ -118,6 +118,13 @@ public async Task<Plan> CreatePlanAsync(string goal)
             plan = new Plan(goal);
         }
 
+        // Planner should not suggest server-url as a parameter.
+        // Server URL should be be parsed from OpenAPI spec or provided by user
+        if (planData.Plan.Parameters.ContainsKey("server-url"))
```

Comment:
There is no need to do a "ContainsKey" check if all you're doing is removing. Dictionary.Remove will not fail if the key does not exist.